openslides/core/static/js/core/base.js

```diff
@@ -39,40 +39,41 @@ angular.module('OpenSlidesApp.core', [
     'DS',
     '$rootScope',
     function (DS, $rootScope) {
-        var url = location.origin + "/sockjs";
-        var socket;
+        var socket = null;
+        var recInterval = null;
+        $rootScope.connected = true;
+
+        var newConnect = function () {
+            socket = new SockJS(location.origin + "/sockjs");
+            clearInterval(recInterval);
+            socket.onopen = function () {
+                $rootScope.connected = true;
+            };
+            socket.onclose = function () {
+                $rootScope.connected = false;
```

Comment:
onclose is called when the connection is closed. There is no difference if the connection is closed by the server, by the client or by some other reasons.

You can add a css-animation, so the red bar is shown 1 second after connected is set to false. So the red bar keeps hidden, if the connection lost is only for 1 second.